src/printer.rs

```diff
@@ -75,85 +93,173 @@ impl<'a> Printer<'a> {
         line_number: usize,
         regions: &[(highlighting::Style, &str)],
     ) -> Result<()> {
-        let decorations = vec![
-            self.print_line_number(line_number),
-            self.print_git_marker(line_number),
-            self.print_line_border(),
-            Some(as_terminal_escaped(
-                &regions,
-                self.config.true_color,
-                self.config.colored_output,
-            )),
-        ];
-
-        let grid_requested = self.config.output_components.grid();
-        write!(
-            self.handle,
-            "{}",
-            decorations
-                .into_iter()
-                .filter_map(|dec| if grid_requested {
-                    Some(dec.unwrap_or_else(|| " ".to_owned()))
-                } else {
-                    dec
-                })
-                .collect::<Vec<_>>()
-                .join(" ")
-        )?;
+        let mut cursor: usize = 0;
+        let mut cursor_max: usize = self.config.term_width;
+
+        // Line decoration.
+        let decorations = self.gen_decorations(line_number);
+        let gutter_width = decorations.len() + decorations.iter().fold(0, |a, x| a + x.size);
+
+        if gutter_width > 0 {
+            cursor_max -= gutter_width;
+            write!(
+                self.handle,
+                "{} ",
+                decorations
+                    .iter()
+                    .map(|seg| seg.text.to_owned())
+                    .collect::<Vec<String>>()
+                    .join(" ")
+            )?;
+        }
+
+        // Grid border.
+        let border = if gutter_width > 0 && self.config.output_components.grid() {
+            self.gen_border()
+        } else {
+            PrintSegment {
+                size: 0,
+                text: "".to_owned(),
+            }
+        };
+
+        cursor_max -= border.size;
+        write!(self.handle, "{}", border.text)?;
+
+        // Line contents.
+        if self.config.output_wrap == OutputWrap::None {
+            let true_color = self.config.true_color;
+            let colored_output = self.config.colored_output;
+
+            write!(self.handle, "{}",
+                   regions.iter()
+                       .map(|&(style, text)| as_terminal_escaped(style, text, true_color, colored_output))
+                       .collect::<Vec<_>>()
+                       .join("")
+            )?;
+        } else {
+            for &(style, text) in regions.iter() {
+                let mut chars = text.chars().filter(|c| *c != '\n');
+                let mut remaining = text.chars().filter(|c| *c != '\n').count();
```

Comment:
You can just clone the chars iterator

Reply:
I did that [in the initial PR](https://github.com/sharkdp/bat/pull/102/commits/4e4110bf5015009d584ed0ed5d09d19eb5393430#diff-041fe12915c4f607e78f57794ffa89bdR136), but it [failed to build on Rust 1.24.0](https://travis-ci.org/sharkdp/bat/jobs/378014200).

Reply:
Huh weird